amazonka/HaskellWorks/Polysemy/Amazonka.hs

```diff
@@ -29,15 +42,26 @@ runReaderAwsEnvDiscover f = do
 
 sendAws :: ()
   => AWS.AWSRequest a
+  => Member (DataLog AwsLogEntry) r
   => Member (Embed m) r
   => Member (Error AWS.Error) r
   => Member (Reader AWS.Env) r
+  => Member Resource r
   => MonadIO m
   => Typeable (AWS.AWSResponse a)
   => Typeable a
   => a
   -> Sem r (AWS.AWSResponse a)
 sendAws req = do
-  envAws <- ask @AWS.Env
-  result <- embed $ liftIO $ runResourceT $ AWS.sendEither envAws req
-  fromEither result
+  tStack <- newTVarIO @[AwsLogEntry] []
+  envAws0 <- ask @AWS.Env
+
+  let logger ::  AWS.LogLevel -> Builder -> IO ()
+      logger ll b = STM.atomically $ STM.modifyTVar tStack (AwsLogEntry ll b:)
```

Comment:
Wouldn't a `TChan` be more suitable?

Reply:
I guess we don't know what the max number of log items to expect.

Reply:
The queue is short live (one per AWS request) so we don't currently bother having any kind of limit.